adapters/algorix/algorix.go

```diff
@@ -0,0 +1,166 @@
+	headers.Add("x-openrtb-version", "2.5")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endPoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, nil
+}
+
+//parseAlgoriXExt parse AlgoriX Ext
+func parseAlgoriXExt(request *openrtb2.BidRequest) *openrtb_ext.ExtImpAlgorix {
+	var extImpAlgoriX openrtb_ext.ExtImpAlgorix
+	for _, imp := range request.Imp {
+		var extBidder adapters.ExtImpBidder
+		err := json.Unmarshal(imp.Ext, &extBidder)
+		if err != nil {
+			continue
+		}
+		err = json.Unmarshal(extBidder.Bidder, &extImpAlgoriX)
+		if err != nil {
+			continue
+		}
+		return &extImpAlgoriX
+	}
+	return nil
+}
+
+// getEndPoint get Endpoint Target
+func (a *adapter) getEndPoint(ext *openrtb_ext.ExtImpAlgorix) (string, error) {
+	endPointParams := macros.EndpointTemplateParams{SourceId: ext.Sid, AccountID: ext.Token}
+	return macros.ResolveMacros(a.EndpointTemplate, endPointParams)
+}
+
+// preProcess process Request check and modify banner imp
+func preProcess(request *openrtb2.BidRequest) {
+	for i := range request.Imp {
+		if request.Imp[i].Banner != nil {
+			banner := *request.Imp[i].Banner
+			if (banner.W == nil || banner.H == nil || *banner.W == 0 || *banner.H == 0) && len(banner.Format) > 0 {
+				firstFormat := banner.Format[0]
+				bannerCopy := *request.Imp[i].Banner
+				bannerCopy.W = &firstFormat.W
+				bannerCopy.H = &firstFormat.H
+				request.Imp[i].Banner = &bannerCopy
+			}
+		}
+	}
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	for _, seatBid := range bidResp.SeatBid {
+		for idx := range seatBid.Bid {
+			mediaType := getBidType(seatBid.Bid[idx].ImpID, internalRequest.Imp)
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatBid.Bid[idx],
+				BidType: mediaType,
+			})
+		}
+	}
+	return bidResponse, nil
+}
+
+// getBidType get Bid Type default: BannerType
+func getBidType(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Native != nil {
+				return openrtb_ext.BidTypeNative
+			}
+			if imp.Video != nil {
+				return openrtb_ext.BidTypeVideo
+			}
+		}
+	}
+	return openrtb_ext.BidTypeBanner
```

Comment:
This is an anti-pattern. Does your bidding server include the bid type in it's response?

Reply:
This is how this function is called:
```
mediaType := getBidType(seatBid.Bid[idx].ImpID, internalRequest.Imp)
```
where first argument is bid from response and second argument is impressions from request. 
this line `if imp.ID == impId {` determines type if bid based on imp type. 

Please advice if you think it's not ok.

Reply:
@SyntaxNode @VeronikaSolovei9 Hello, this func is determines BidType in the bidResponse.Bids. In my understanding, this func is similar in many adapters(base OpenRtb 2.5 protocol). can you tell me which adapter func getMediaTypeForImp(getBidType) is standard, thx. we check Inmobi, smartyads, yeahmobi, BrightMountainMedia and others adapter is the same as func getMediaTypeForImp(getBidType).

I see the new Adapters E-Volution, has the same problem? So the newest solution is add bid type in bid.ext?

So I support add the sample func getBidType in Doc, in this doc: https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html, we can't find getMediaTypeForBid() how to implement.